### PR TITLE
Updated protobufs to reflect Mesos 0.23.0.

### PR DESCRIPTION
**Pull request by karya0**
A `make all` succeeds but `make all tests` fails with some errors that are beyond my comprehension.

### CODE REVIEW

**Comment by vieux** [comment]
@karya0 you forgot to update the `https://github.com/gogo/protobuf` dependency.

`godep update github.com/gogo/protobuf/proto` should do the trick

**Comment by karya0** [comment]
Thanks @vieux. The build succeeds but the tests are failing now :-(. Looks like they have been broken for a while now.

**Comment by vieux** [comment]
@karya0 the failure in `github.com/mesos/mesos-go/auth/sasl` doesn't seem too bad, probably just a few mock to update.

The one in `github.com/mesos/mesos-go/executor` seems to be quite bad though, looks like a mandatory field in the protobuf isn't set.

```
E0806 15:57:04.740347   26399 messenger.go:367] Failed to unmarshal message &{slave(1)@127.0.0.1:51175 mesos.internal.ExecutorRegisteredMessage executor_info:<> framework_id:<value:"some-framework-id-uuid" > framework_info:<user:"test" name:"test-framework" id:<value:"some-framework-id-uuid" > > slave_id:<value:"some-slave-id-uuid" > slave_info:<hostname:"localhost" >  [18 0 26 24 10 22 115 111 109 101 45 102 114 97 109 101 119 111 114 107 45 105 100 45 117 117 105 100 34 48 10 4 116 101 115 116 18 14 116 101 115 116 45 102 114 97 109 101 119 111 114 107 26 24 10 22 115 111 109 101 45 102 114 97 109 101 119 111 114 107 45 105 100 45 117 117 105 100 42 20 10 18 115 111 109 101 45 115 108 97 118 101 45 105 100 45 117 117 105 100 50 11 10 9 108 111 99 97 108 104 111 115 116]}: proto: required field "ExecutorInfo.executor_id" not set
```

**Comment by vieux** [comment]
@karya0 look like the `Command` field in `ExecutorInfo` is required and in the tests it's set to nil: https://github.com/mesos/mesos-go/blob/master/executor/executor_intgr_test.go#L188

**Comment by karya0** [comment]
Thanks @vieux! Fixed ExecutorInfo.command and that tests seems to work now. Just need to figure out the mock stuff now (given that I don't speak any GO ;-) ).

**Comment by vieux** [comment]
@karya0 GO or not, it seems to me that is rather scary to update the tests to have them pass. The tests are passing on master without having to specify a command. Something changed in this PR.

**Comment by vieux** [comment]
the mesos-swarm adaptor is broken with this PR so I think something changed regarding protobuf parsing in this PR /cc @tnachen

**Comment by tsenart** [comment]
I'll take a look at this today. Hang in there :-)

**Comment by tsenart** [comment]
I did some research and it seems you need to change the code a little more to support this Protobufs upgrade. It seems we have [no handler for `AuthenticationStartMessage`](https://github.com/karya0/mesos-go/blob/mesos-0.23.0/auth/sasl/authenticatee.go#L204) which I believe is the cause for the test failure.
```
panic:

mock: Unexpected Method Call
-----------------------------

Send(string,*upid.UPID,*mesosproto.AuthenticationStartMessage)
		0: mock.Anything
		1: serv@a.com:123
		2: mechanism:"CRAM-MD5"

The closest call I have is:

Send(string,*upid.UPID,*mesosproto.AuthenticationStepMessage)
		0: mock.Anything
		1: serv@a.com:123
		2: data:"foo cc7fd96cd80123ea844a7dba29a594ed"


goroutine 8 [running]:
github.com/stretchr/testify/mock.(*Mock).Called(0xc208074000, 0xc20803ab10, 0x3, 0x3, 0x0, 0x0, 0x0)
	/Users/tomas/go/src/github.com/stretchr/testify/mock/mock.go:266 +0x44e
github.com/mesos/mesos-go/auth/sasl.(*MockTransport).Send(0xc20802c080, 0x69d240, 0xc20803b9b0, 0xc20803b5f0, 0x69d6f8, 0xc20802a340, 0x0, 0x0)
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee_test.go:23 +0x19e
github.com/mesos/mesos-go/auth/sasl.(*authenticateeProcess).mechanisms(0xc208056120, 0x69d240, 0xc20803b9b0, 0xc20803b5f0, 0x69d280, 0xc20803b980)
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee.go:291 +0x74c
github.com/mesos/mesos-go/auth/sasl.*authenticateeProcess.(github.com/mesos/mesos-go/auth/sasl.mechanisms)·fm(0x69d240, 0xc20803b9b0, 0xc20803b5f0, 0x69d280, 0xc20803b980)
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee.go:204 +0x59
github.com/mesos/mesos-go/auth/sasl.func·006(0xc20803b5f0, 0x69d280, 0xc20803b980)
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee.go:194 +0x157
github.com/mesos/mesos-go/messenger.(*MockedMessenger).recvLoop(0xc208074000)
	/Users/tomas/go/src/github.com/mesos/mesos-go/messenger/mocked_messenger.go:97 +0x1d9
created by github.com/mesos/mesos-go/messenger.(*MockedMessenger).Start
	/Users/tomas/go/src/github.com/mesos/mesos-go/messenger/mocked_messenger.go:69 +0x44

goroutine 1 [chan receive]:
testing.RunTests(0x4a09c0, 0x5c5d20, 0x1, 0x1, 0xc20800ae01)
	/usr/local/Cellar/go/1.4.2/libexec/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc208052320, 0x5d5560)
	/usr/local/Cellar/go/1.4.2/libexec/src/testing/testing.go:485 +0x6c
main.main()
	github.com/mesos/mesos-go/auth/sasl/_test/_testmain.go:52 +0x1d5

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x5cd8c0)
	/Users/tomas/go/src/github.com/golang/glog/glog.go:882 +0x78
created by github.com/golang/glog.init·1
	/Users/tomas/go/src/github.com/golang/glog/glog.go:410 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/Cellar/go/1.4.2/libexec/src/runtime/asm_amd64.s:2232 +0x1

goroutine 7 [select]:
github.com/mesos/mesos-go/auth/sasl.func·004(0x69d240, 0xc20803b7a0, 0x69d1a0, 0xc20801e440, 0x0, 0x0)
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee.go:123 +0x237
github.com/mesos/mesos-go/auth.AuthenticateeFunc.Authenticate(0xc20801e500, 0x69d160, 0xc20800acd0, 0x69d1a0, 0xc20801e440, 0x0, 0x0)
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/interface.go:26 +0x5d
github.com/mesos/mesos-go/auth/sasl.TestAuthticatee_validLogin(0xc208056090)
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee_test.go:94 +0x5c2
testing.tRunner(0xc208056090, 0x5c5d20)
	/usr/local/Cellar/go/1.4.2/libexec/src/testing/testing.go:447 +0xbf
created by testing.RunTests
	/usr/local/Cellar/go/1.4.2/libexec/src/testing/testing.go:555 +0xa8b

goroutine 9 [chan receive]:
github.com/mesos/mesos-go/auth/sasl.func·005()
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee.go:173 +0x5e
created by github.com/mesos/mesos-go/auth/sasl.(*authenticateeProcess).startTransport
	/Users/tomas/go/src/github.com/mesos/mesos-go/auth/sasl/authenticatee.go:177 +0xf0
FAIL	github.com/mesos/mesos-go/auth/sasl	0.011s
```

**Comment by tnachen** [comment]
I'd like to get this fixed since one of our unit test in Mesos is using a go executor and failing without 0.23.0  proto upgrade.

**Comment by tnachen** [comment]
Also docker swarm will need this to work with 0.23.0 too!

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	"Deps": [
 		{
 			"ImportPath": "github.com/gogo/protobuf/proto",
-			"Rev": "ab6cea4a44ef42b748cd88d2d372047b75806e0c"
```
Comment (jdef) [inline Godeps/Godeps.json]: why was this updated? bugfix?

File: mesosproto/messages.proto [hunk 2]
```diff
@@ -426,28 +437,9 @@ message TaskHealthStatus {
...
-
-      // Module-specific parameters.
-      repeated Parameter parameters = 2;
-    }
-
-    repeated Module modules = 3;
-  }
-
-  repeated Library libraries = 1;
+/**
```
Comment (jdef) [inline mesosproto/messages.proto]: it looks like we lost the modules protos. do we care?

Comment (tnachen) [inline mesosproto/messages.proto]: I don't think we care about modules in the mesos-go codebase

File: Godeps/Godeps.json (hunk 1 above)
Comment (karya0) [inline Godeps/Godeps.json]: I think so. @vieux had initially suggested to run `godep update github.com/gogo/protobuf/proto`.

**Comment by jdef** [comment]
lgtm :ship:

**Pull request merged**